src/cephadm/cephadm

```diff
@@ -9375,6 +9436,10 @@ def _get_parser():
         '--daemon-id',
         help='daemon id for agent')
 
+    parser_agent = subparsers.add_parser(
+        'disk-rescan', help='rescan all HBAs to detect new/removed devices')
+    parser_agent.set_defaults(func=command_rescan_disks)
+
```

Comment:
seems ok, but wondering if this really should be a ceph-volume command instead?

Reply:
There's a couple of  reason I placed it here 
- running ceph-volume requires the container to be instantiated - that's time. 1.5-2s on the physical machines I tested with
- it doesn't need anything in ceph-volume
- its not managing or changing any of the devices
- it's not going to be an async thing - its going to rely on responsiveness since the admin will issue the command for a rescan ad-hoc. The longer it takes the longer we tie up the cli thread
- it provides a more responsive UX

Reply:
> seems ok, but wondering if this really should be a ceph-volume command instead?

yes, it could be, but in my opinion it doesn't have to.
If we implement this in `ceph-volume`, users will be bound to run it from within a container which is definitely not convenient and not needed for such a command/operation.